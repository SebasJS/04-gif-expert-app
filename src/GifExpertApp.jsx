import { useState } from "react";
import { AddCategory } from "./components/AddCategory";


export const GifExpertApp = () => {

  const [ categories, setCategories ] = useState([ 'One Puch', 'Dragon Ball']);

  const onAddCategory = () => {
    /*No se recomienda el uso del push, es mejor utilizar el operador ... */
    setCategories([...categories, 'Avengers']);

  }

  console.log( categories );
  return (
    <>
      {/* Titulo */}
        
        <h1> GifExpertApp</h1>

      {/* Input */}
        < AddCategory />

      {/* Listado de Items */}
        {/* Item */}
        <ol>
          { categories.map( ( category ) => {
            {/* En este tipo de creacion de listas se le debe dar una key al item esta debe ser unica para el elemento */}
            return <li key={ category }> { category } </li>
          }) }
        </ol>
    </>
  )
}

