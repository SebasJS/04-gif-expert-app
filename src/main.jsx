import React from 'react'
import ReactDOM from 'react-dom/client'
import { GifExpertApp } from './GifExpertApp'
import './styles.css';

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    {/* Titulo */}
    <GifExpertApp/>
    {/* Input */}

    {/* Listado de Items */}

  </React.StrictMode>
)
