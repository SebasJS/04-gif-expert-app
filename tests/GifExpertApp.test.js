import { GifExpertApp } from '../src/GifExpertApp'
import { fireEvent, render, screen } from '@testing-library/react';

describe('Pruebas de GifExpertApp', () => {
    
    const newCategory = 'Dragon Ball'

    test('debe de hacer match con el snatshot', () => {
        
        const { container } = render( < GifExpertApp /> )
        expect(container).toMatchSnapshot()

    });
    test('debe de agregar la categoria', () => {
        
        render( < GifExpertApp /> )
        
        const input = screen.getByRole('textbox');
        const form = screen.getByRole('form');
        
        fireEvent.input(input, { target: { value: newCategory } } );
        fireEvent.submit(form);
        
        screen.debug();
        expect(screen.getAllByRole('heading', {level:3}).length).toBe(1);

    });

    test('no debe ingresar el valor si ya existe', () => {
        
        render( < GifExpertApp /> )
        
        const input = screen.getByRole('textbox');
        const form = screen.getByRole('form');
        
        fireEvent.input(input, { target: { value: newCategory } } );
        fireEvent.submit(form);
        
        fireEvent.input(input, { target: { value: newCategory } } );
        fireEvent.submit(form);

        expect(screen.getAllByRole('heading', {level:3}).length).toBe(1);
        screen.debug();        

    });
});