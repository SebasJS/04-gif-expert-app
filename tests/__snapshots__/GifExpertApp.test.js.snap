// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de GifExpertApp debe de hacer match con el snatshot 1`] = `
<div>
  <h1>
     GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
  <ol />
</div>
`;
